jenkins_remote/jenkins_remote.py

```
@@ -0,0 +1,145 @@
+        f"{JENKINS_URL}/{CI_MONITOR_JOBS}/{job_url}/buildWithParameters?{params}"
+    )
+    r = requests.post(
+        ci_monitor_url,
+        auth=(os.getenv("USER"), os.getenv("JENKINS_API_TOKEN")),
+        verify="/etc/certs/ipa.crt",
+    )
+    if r.status_code == 201:
+        nbuild = _get_jenkins_build_number(r.headers["location"])
+        print(f"triggered job: {nbuild} to report CI failures successfully")
+
+
+def install_cluster(args):
+    job_url = "Flexy-install"
+    version = args.ocp_version
+    install_type = "ipi" if args.ipi else "upi"
+    provider = args.provider
+    path_to_profile = f"functionality-testing/aos-{version.replace('.', '_')}/{install_type}-on-{provider}/versioned-installer-{args.profile}"
+    git_localtion = f"https://gitlab.cee.redhat.com/aosqe/flexy-templates/-/raw/master/{path_to_profile}"
+
+    r = requests.get(git_localtion, verify="/etc/certs/ipa.crt")
+    if r.status_code != 200:
+        raise Exception(f"profile {path_to_profile} not found")
+    jenkins_path_to_profile = f"private-templates/{path_to_profile}"
+
+    print(f"successfully found profile {path_to_profile}")
+
+    flexy_job_url = f"{JENKINS_URL}/{OCP_COMMON_JOBS}/{job_url}/buildWithParameters"
+
+    r = requests.post(
+        flexy_job_url,
+        auth=(os.getenv("USER"), os.getenv("JENKINS_API_TOKEN")),
+        verify="/etc/certs/ipa.crt",
+        params={
+            "INSTANCE_NAME_PREFIX": args.name,
+            "VARIABLES_LOCATION": jenkins_path_to_profile,
+        },
+    )
+    if r.status_code == 201:
+        nbuild = _get_jenkins_build_number(r.headers["location"])
+        print(f"triggered job to install cluster successfully with build id: {nbuild}")
+
+
+def _get_jenkins_build_number(location):
+    nbuild_resp = requests.get(f"{location}/api/json", verify="/etc/certs/ipa.crt")
+    if nbuild_resp.status_code == 200:
+        response = nbuild_resp.json()
+        return response["executable"]["number"]
+
```

Maybe raise an exception here if you don't get a 200 status code back?